internal/vault/client_test.go

```diff
@@ -747,3 +747,64 @@ func Test_defaultClient_Read(t *testing.T) {
 		})
 	}
 }
+
+func Test_defaultClient_Close(t *testing.T) {
+	handlerFunc := func(t *testHandler, w http.ResponseWriter, req *http.Request) {
+		if req.Method != http.MethodPut {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+	}
+
+	tests := []struct {
+		name           string
+		revoke         bool
+		handler        *testHandler
+		expectRequests int
+		expectParams   []map[string]interface{}
+		expectPaths    []string
+	}{
+		{
+			name:   "ensure-closed",
+			revoke: false,
+			handler: &testHandler{
+				handlerFunc: handlerFunc,
+			},
+		},
+		{
+			name:   "ensure-closed-with-revoke",
+			revoke: true,
+			handler: &testHandler{
+				handlerFunc: handlerFunc,
+			},
+			expectPaths:    []string{"/v1/auth/token/revoke-self"},
+			expectRequests: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config, l := NewTestHTTPServer(t, tt.handler.handler())
```

Comment:
Looks like these tests use the same handlerFunc:

```suggestion
			h := &testHandler{
				handlerFunc: handlerFunc,
			}
			config, l := NewTestHTTPServer(t, h.handler())
```

Reply:
Fixed with a20daa7354a913236fd6d781ad9dad28afee0831